cmd/ltagent/init.go

```diff
@@ -107,63 +74,95 @@ func RunInitCmdF(cmd *cobra.Command, args []string) error {
-		return err
-	}
-	err = store.SetUser(&model.User{
-		Email:    config.ConnectionConfiguration.AdminEmail,
-		Password: config.ConnectionConfiguration.AdminPassword,
-	})
-	if err != nil {
-		return err
-	}
-
-	admin := userentity.New(userentity.Setup{Store: store}, ueConfig)
-
-	start := time.Now()
-
-	if err = admin.Login(); err != nil {
-		return err
+	if err := defaults.Validate(*config); err != nil {
+		return fmt.Errorf("could not validate configuration: %w", err)
 	}
 
-	if err = createTeams(admin, numTeams); err != nil {
+	userPrefix, err := cmd.Flags().GetString("user-prefix")
+	if err != nil {
 		return err
 	}
 
-	if err = createChannels(admin, numChannels); err != nil {
+	if ok, err := initDone(config.ConnectionConfiguration.ServerURL, userPrefix); err != nil {
 		return err
+	} else if ok {
+		mlog.Warn("init already done")
+		return nil
+	}
+
+	seed := memstore.SetRandomSeed()
+	mlog.Info(fmt.Sprintf("random seed value is: %d", seed))
+
+	// http.Transport to be shared amongst all clients.
+	transport := &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   1 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).DialContext,
+		MaxConnsPerHost:       100,
+		MaxIdleConns:          100,
+		MaxIdleConnsPerHost:   100,
+		ResponseHeaderTimeout: 10 * time.Second,
+		IdleConnTimeout:       90 * time.Second,
+		TLSHandshakeTimeout:   1 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
+
+	genConfig := config.InstanceConfiguration
+
+	newControllerFn := func(id int, status chan<- control.UserStatus) (control.UserController, error) {
+		ueConfig := userentity.Config{
+			ServerURL:    config.ConnectionConfiguration.ServerURL,
+			WebSocketURL: config.ConnectionConfiguration.WebSocketURL,
+			Username:     fmt.Sprintf("%s-%d", userPrefix, id),
+			Email:        fmt.Sprintf("%s-%d@example.com", userPrefix, id),
+			Password:     "testPass123$",
+		}
+		store, err := memstore.New(&memstore.Config{
+			MaxStoredPosts:          500,
+			MaxStoredUsers:          1000,
+			MaxStoredChannelMembers: 1000,
+			MaxStoredStatuses:       1000,
+		})
+		if err != nil {
+			return nil, err
+		}
+		ueSetup := userentity.Setup{
+			Store:     store,
+			Transport: transport,
+		}
+		ue := userentity.New(ueSetup, ueConfig)
+		return gencontroller.New(id, ue, &genConfig, status)
 	}
 
-	if err = createTeamAdmins(admin, numTeamAdmins, config); err != nil {
-		return err
+	config.UsersConfiguration.InitialActiveUsers = 0
+	config.UserControllerConfiguration.RatesDistribution = []loadtest.RatesDistribution{
+		{
+			Rate:       0.2,
+			Percentage: 1.0,
+		},
 	}
 
-	if _, err = admin.Logout(); err != nil {
-		return err
+	lt, err := loadtest.New(config, newControllerFn)
+	if err != nil {
+		return fmt.Errorf("error while initializing loadtest: %w", err)
 	}
 
-	mlog.Info("done", mlog.String("elapsed", time.Since(start).String()))
-
-	return nil
+	return genData(lt, 50)
```

Comment:
These values (rate and number of users) come from various benchmarks that I ran. I was able to get decent times with these on different configurations.